src/internal_events/redis.rs

```diff
@@ -0,0 +1,70 @@
+use super::InternalEvent;
+use metrics::counter;
+
+#[derive(Debug)]
+pub struct RedisEventReceived {
+    pub byte_size: usize,
+}
+
+impl InternalEvent for RedisEventReceived {
+    fn emit_logs(&self) {
+        trace!(message = "Received one event.", rate_limit_secs = 10);
+    }
+
+    fn emit_metrics(&self) {
+        counter!("processed_bytes_total", self.byte_size as u64);
```

Comment:
I think we should not emit this metric. This should only be emitted when the event is sent by `RedisEventSent` otherwise it'll double count the bytes for each message.